kroma-validator/challenger.go

```diff
@@ -734,7 +734,26 @@ func (c *Challenger) IsOutputFinalized(ctx context.Context, outputIndex *big.Int
 func (c *Challenger) GetChallenge(ctx context.Context, outputIndex *big.Int, challenger common.Address) (bindings.TypesChallenge, error) {
 	cCtx, cCancel := context.WithTimeout(ctx, c.cfg.NetworkTimeout)
 	defer cCancel()
-	return c.colosseumContract.GetChallenge(optsutils.NewSimpleCallOpts(cCtx), outputIndex, challenger)
+
+	challenge, err := c.colosseumContract.Challenges(optsutils.NewSimpleCallOpts(cCtx), outputIndex, challenger)
+	if err != nil {
+		return bindings.TypesChallenge{}, fmt.Errorf("failed to fetch challenge data: %w", err)
+	}
+
+	segments, err := c.colosseumContract.GetSegments(optsutils.NewSimpleCallOpts(cCtx), outputIndex, challenger)
```

Comment:
Even though `challenges` mapping is declared to public variable, the getter function auto generated by solidity compiler returns the `Challenge` struct excluding dynamic array field `segments`. So needed to add additional function to get `segments`. Also a function returning single field rather than whole struct has less code size.

Reply:
Oh I see. I didn't know that.

Reply:
Do you mean "abigen"?